python_modules/dagster/dagster/core/definitions/resource_requirement.py

```diff
@@ -197,3 +199,30 @@ def ensure_requirements_satisfied(
             raise DagsterInvalidDefinitionError(
                 f"{requirement.describe_requirement()} was not provided{mode_descriptor}. Please provide a {str(requirement.expected_type)} to key '{requirement.key}', or change the required key to one of the following keys which points to an {str(requirement.expected_type)}: {requirement.keys_of_expected_type(resource_defs)}"
             )
+
+
+def get_and_validate_transitive_resource_dependencies(
```

Comment:
I find this function a little bit confusing to grok, because it's kind of doing two things at the same time. Would it make sense to split them out into two steps?

Reply:
this is a case where we cannot do that unless we want to create a secondary error state, as in order to get the next set of dependencies we need to validate the previous set.

Reply:
Got it 👍

Reply:
hmm upon further thinking I don't think what I say here is true ^. Working to split these out.